### ISSUE TITLE
"rest_get" and "mpirc" snippets provided in incorrect contexts

### ISSUE THREAD

**Issue by dessina-devasia** [OpenLiberty/liberty-tools-intellij#695]
During the FVT for version 24.0.3-SNAPSHOT on a Windows 11 environment, an issue was encountered with the identification of 'rest_get' and 'mpirc' snippets under certain conditions in Java files. Specifically, the issue arises in the following scenario:

In the space before a class declaration (empty space).
For reference, I've attached a screenshot below showcasing the same.

![image](https://github.com/OpenLiberty/liberty-tools-intellij/assets/143582034/6e280158-2ee8-4b42-b773-2e21f50be6c4)

This issue has been reported to lsp4mp as issue https://github.com/eclipse/lsp4mp/issues/402

**Comment by TrevCraw** [comment]
Most likely an issue with LSP4MP incorrectly setting the context value for these snippets

**Comment by turkeylurkey** [comment]
`mpirc` is the only snippet marked as a `field` and `rest_get` is the only snippet marked as a `method.` LSP4MP needs to improve its handling of these two types of snippets.

**Comment by turkeylurkey** [comment]
We are discussing what to do in lsp4mp, one alternate proposal suggested.

**Comment by vaisakhkannan** [comment]
I can confirm that the issue is present in all three IDEs.

Screenshot from VSCode

<img width="680" alt="Image" src="https://github.com/user-attachments/assets/09c23f39-7e38-4468-b8ee-26adaefa1135" />

Screenshot from Eclipse

<img width="756" alt="Image" src="https://github.com/user-attachments/assets/5c008b6d-4ead-45bb-8294-092502604431" />

**Comment by turkeylurkey** [comment]
I received some direction from Red Hat concerning this fix and I wrote a new fix. I put this fix in the existing lsp4mp pull request and requested a review. Once that fix is merged and a new release is available we can merge the fix attached to this issue.

**Comment by turkeylurkey** [comment]
The lsp4mp pull request referenced above is merged so we just need a new release to make use of the fix. Draft pull request https://github.com/OpenLiberty/liberty-tools-intellij/pull/1285 is opened. We just need lsp4mp release details to complete the fix and then this pull request can be reviewed and merged.

**Comment by anusreelakshmi934** [comment]
Verified this issue no longer occurs in the branch [lsp4mp-0.17.0-support](https://github.com/OpenLiberty/liberty-tools-intellij/tree/lsp4mp-0.17.0-support)

<img width="477" height="363" alt="Image" src="https://github.com/user-attachments/assets/3d257c5d-5d2d-4cf7-b052-2304dbb7b0ff" />

The fix for this https://github.com/OpenLiberty/liberty-tools-intellij/pull/1718 has been merged into the feature branch.
